src/app/Rules/MatchExpressionEvaluator.tsx

```diff
@@ -0,0 +1,141 @@
+ */
+import * as React from 'react';
+import { CodeBlock, CodeBlockCode, Label, Stack, StackItem, Text, ValidatedOptions } from '@patternfly/react-core';
+import { ServiceContext } from '@app/Shared/Services/Services';
+import { useSubscriptions } from '@app/utils/useSubscriptions';
+import { Target } from '@app/Shared/Services/Target.service';
+import { TargetSelect } from '@app/TargetSelect/TargetSelect';
+import { NoTargetSelected } from '@app/TargetView/NoTargetSelected';
+import {CheckCircleIcon, InfoCircleIcon, QuestionCircleIcon, WarningTriangleIcon} from '@patternfly/react-icons';
+
+export interface MatchExpressionEvaluatorProps {
+  matchExpression?: string;
+}
+
+export const MatchExpressionEvaluator: React.FunctionComponent<MatchExpressionEvaluatorProps> = (props) => {
+  const context = React.useContext(ServiceContext);
+  const addSubscription = useSubscriptions();
+  const [target, setTarget] = React.useState(undefined as Target | undefined);
+  const [valid, setValid] = React.useState(ValidatedOptions.default);
+
+  React.useEffect(() => {
+    addSubscription(
+      context.target.target().subscribe(setTarget)
+    );
+  }, [context, context.target]);
+
+  React.useEffect(() => {
+    if (!props.matchExpression || !target?.connectUrl) {
+      setValid(ValidatedOptions.default);
+      return;
+    }
+    try {
+      const f = new Function('target', `return ${props.matchExpression}`);
+      const res = f(target);
+      if (typeof res === 'boolean') {
+        setValid(res ? ValidatedOptions.success : ValidatedOptions.warning);
+        return;
+      }
+      setValid(ValidatedOptions.error);
+      return;
+    } catch (err) {
+      setValid(ValidatedOptions.error);
+      return;
+    }
+  }, [target, props.matchExpression]);
+
+  const statusLabel = React.useMemo(() => {
+    switch (valid) {
+      case ValidatedOptions.success:
+        return (<Label color="green" icon={<CheckCircleIcon />}>Expression Matches Selected Target</Label>);
+      case ValidatedOptions.warning:
+        return (<Label color="orange" icon={<QuestionCircleIcon />}>Expression Valid, No Match</Label>);
+      case ValidatedOptions.error:
+        return (<Label color="red" icon={<WarningTriangleIcon />}>Invalid Match Expression</Label>);
+      default:
+        return (<Label color="grey" icon={<InfoCircleIcon />}>No Match Expression</Label>);
+    }
+  }, [valid]);
+
+  const exampleExpression = React.useMemo(() => {
+    let body: string;
+    if (!target || !target?.alias || !target?.connectUrl) {
+      body = 'true';
+    } else {
+      body = `target.alias == '${target?.alias}' || target.annotations.cryostat['PORT'] == ${target?.annotations?.cryostat['PORT']}`;
+    }
+    return (<>
+      <CodeBlock>
+        <CodeBlockCode>
+          { JSON.stringify(body, null, 2) }
+        </CodeBlockCode>
+      </CodeBlock>
+    </>);
+  },[target]);
+
+  return (<>
+    <Stack hasGutter>
+      <StackItem>
+        <TargetSelect isCompact />
```

Comment:
Is it necessary to provide the `isCompact` prop to `TargetSelect` anymore since it looks like the component leaves the decision of whether it is compact or not to its parent component?

Reply:
Well spotted, thanks. I've removed the unused property.